TypeCobol/Compiler/Scanner/AbstractScanner.cs

```diff
@@ -62,5 +71,299 @@ protected Token ScanOneChar(int startIndex, TokenType tokenType)
+            // ’THIS IS RIGHT’
+            // "THIS IS RIGHT"
+            // ’THIS IS WRONG"
+            // You can use apostrophes or quotation marks as the literal delimiters independent
+            // of the APOST/QUOTE compiler option.
+            // Any punctuation characters included within an alphanumeric literal are part of the
+            // value of the literal.
+            // The maximum length of an alphanumeric literal is 160 bytes. The minimum length
+            // is 1 byte.
+
+            // consume opening delimiter
+            char delimiter = line[currentIndex];
+            currentIndex++;
+
+            // consume all chars until we encounter one occurrence (and only one) of the delimiter 
+            bool closingDelimiterFound = false;
+            bool usingVirtualSpaceAtEndOfLine = false;
+            StringBuilder sbValue = new StringBuilder();
+            do
+            {
+                for (; currentIndex <= lastIndex && line[currentIndex] != delimiter; currentIndex++)
+                {
+                    char currentChar = line[currentIndex];
+                    sbValue.Append(currentChar);
+                }
+                // delimiter found before the last character of the line
+                if (currentIndex < lastIndex)
+                {
+                    // continue in case of a double delimiter
+                    if (line[currentIndex + 1] == delimiter && !(multiStringConcatBitPosition?.Get(currentIndex + 1) ?? false))
+                    {
+                        // consume the two delimiters
+                        currentIndex += 2;
+                        // append one delimiter to the literal value
+                        sbValue.Append(delimiter);
+                    }
+                    // stop in case of a simple delimiter
+                    else
+                    {
+                        // consume closing delimiter
+                        currentIndex++;
+                        closingDelimiterFound = true;
+                    }
+                }
+                // delimiter found on the last character of the line
+                else if (currentIndex == lastIndex)
+                {
+                    // consume closing delimiter
+                    currentIndex++;
+                    closingDelimiterFound = true;
+                    usingVirtualSpaceAtEndOfLine = true;
+                }
+            } while (currentIndex <= lastIndex && !closingDelimiterFound);
+
+            // create an alphanumeric literal token
+            int endIndex = (currentIndex > lastIndex) ? lastIndex : currentIndex - 1;
+            Token token = new Token(tokenType, startIndex, endIndex, usingVirtualSpaceAtEndOfLine, tokensLine, true, closingDelimiterFound, delimiter);
+
+            // compute the value of the literal, depending on the exact literal type            
+            AlphanumericLiteralTokenValue value=null;
```

Comment:
It is not required to initialize to null here, check the part for graphic strings. You should create the value no matter the kind ('u' or 'g')